firmware/application/apps/ui_level.cpp

```diff
@@ -191,15 +223,23 @@ void LevelView::on_statistics_update(const ChannelStatistics& statistics) {
             .foreground = {255, 255, 255},
         };
         freq_stats_rx.set_style(&style_freq_stats_rx);
+        freq_stats_rx.set("RxSat: " + to_string_dec_uint(rx_sat) + "%");
     }
 
 } /* on_statistic_updates */
 
 size_t LevelView::change_mode(freqman_index_t new_mod) {
     field_bw.on_change = [this](size_t n, OptionsField::value_t) { (void)n; };
 
+    radio_mode = new_mod;
+
+    audio::output::stop();
+    receiver_model.disable();
+    baseband::shutdown();
+
     switch (new_mod) {
         case AM_MODULATION:
+            audio_sampling_rate = audio::Rate::Hz_12000;
             freqman_set_bandwidth_option(new_mod, field_bw);
             baseband::run_image(portapack::spi_flash::image_tag_am_audio);
```

Comment:
You'll need to add support to each baseband process (proc_am_audio, proc_nfm_audio, proc_wfm_audio, and proc_capture) for message::ID:AudioBeep (see example in earlier comment).

Reply:
I only want beep in capture spectrum mode